test/after.test.js

```diff
@@ -0,0 +1,172 @@
+'use strict';
+
+// external files
+var assert = require('chai').assert;
+var bunyan = require('bunyan');
+var restify = require('restify');
+var _ = require('lodash');
+
+// local files
+var clients = require('../lib');
+
+
+describe('`after` event', function () {
+
+    var SERVER;
+    var STRINGCLIENT = clients.createStringClient({
+        url: 'http://localhost:3000/',
+        requestTimeout: 100
+    });
+    var JSONCLIENT = clients.createJsonClient({
+        url: 'http://localhost:3000/',
+        requestTimeout: 100
+    });
+    var LOG = bunyan.createLogger({
+        name: 'clientlog'
+    });
+
+    beforeEach(function (done) {
+        SERVER = restify.createServer({
+            name: 'unittest',
+            log: LOG
+        });
+        SERVER.get('/200', function (req, res, next) {
+            res.send(200, { hello: 'world' });
+            return next();
+        });
+        SERVER.get('/500', function (req, res, next) {
+            res.send(500, { empty: 'world' });
+            return next();
+        });
+        SERVER.get('/timeout', function (req, res, next) {
```

Comment:
Do we even need that route if we use an unresolvable IP to trigger a timeout error?

Reply:
The unresolvable IPs are used for connection timeouts tests, but there are a few request timeout tests that require a "latent" server.

Reply:
Ah right, I missed that we have two different errors, once for _connection_ timeouts and one for _request_ timeouts.